react/Gallery.js

```diff
@@ -1,34 +1,104 @@
 import React from 'react'
-import { PropTypes } from 'prop-types'
 import classNames from 'classnames'
+import { List, WindowScroller, AutoSizer, CellMeasurer, CellMeasurerCache } from 'react-virtualized'
+import { PropTypes } from 'prop-types'
+
+import { withRuntimeContext, NoSSR } from 'render'
 
 import { productShape } from './constants/propTypes'
 import GalleryItem from './components/GalleryItem'
 
 /**
  * Canonical gallery that displays a list of given products.
  */
-const Gallery = ({ products, summary, layoutMode }) => {
-  const classes = classNames('vtex-gallery pa3 bn', {
-    'vtex-gallery--two-columns': layoutMode === 'small',
+
+const Gallery = ({
+  products,
+  summary,
+  layoutMode,
+  runtime: { hints: { mobile } },
+  itemWidth }) => {
+  const cache = new CellMeasurerCache({
+    defaultHeight: 305,
```

Comment:
Fixed value?

Reply:
Its an estimate to the initial unmeasured rows Height, but its not required